ci/linux-debian.Dockerfile

```diff
@@ -23,6 +25,33 @@ RUN apt-get update && apt-get install --no-install-recommends -y \
         sagemath
 
 WORKDIR /root
+
+# Build and install gcc snapshot
+ARG GCC_SNAPSHOT_VERSION=gcc-14-20230702
```

Comment:
This will break at the weekend, when the LATEST-14 symlink no longer points to the 20230702 snapshot. It will point to a different date, and so the name of the tarball will be different.

Reply:
You could replace the LATEST-14 directory with 14-20230702 which will not be invalidated every week, but then your CI will just be using the same snapshot forever.

Reply:
Or just use the ones built for this purpose 😄

Reply:
It also seems quite wasteful to keep building the same snapshots again and again, but that is on brand for bitcoin.

Reply:
> This will break at the weekend, when the LATEST-14 symlink no longer points to the 20230702 snapshot. It will point to a different date, and so the name of the tarball will be different.

Oh sure...

> You could replace the LATEST-14 directory with 14-20230702 which will not be invalidated every week, but then your CI will just be using the same snapshot forever.

Indeed. Hm. The ability to pin a specific snapshot is probably useful. This means we'd need to update the dockerfile regularly (e.g., once per month) with a PR. I think that's okay. It's just a bit inconsistent if we don't do the same for the clang snapshots...

For now, I simply replaced this with a version that downloads the latest snapshot. (Using wget's recursion feature that recurses into downloading links.) I think any approach will be good enough as an initial attempt. We'll need to see how often we run into problems. 


> Or just use the ones built for this purpose smile

You mean those at https://jwakely.github.io/pkg-gcc-latest/ ?

Reply:
> It also seems quite wasteful to keep building the same snapshots again and again,

Note that our CI infrastructure builds this once as a docker image, and then every CI jobs runs in a new container using this prebuilt image, which has the GCC snapshot preinstalled then. The docker image is only rebuilt if we update the Dockerfile, or otherwise trigger a rebuild manually. That's a pretty neat feature of Cirrus CI, see https://cirrus-ci.org/guide/docker-builder-vm/#dockerfile-as-a-ci-environment